daml-lf/repl/src/main/scala/com/digitalasset/daml/lf/repl/testing/Main.scala

```diff
@@ -295,6 +296,7 @@ object Repl {
       if (needParens) s"($s)" else s
 
     def prettyType(t0: Type, prec: Int = precTForall): String = t0 match {
+      case TSynApp(_, _) => "<TODO: #3616,typeSynApp>"
```

Comment:
This pretty printer is not very important. 
It should be fine to inline `TSynApp(tyCon, args)` as `TApp(TyCon(tycon), args)`

Reply:
I filled in the PP code for this case